pkgs/development/libraries/qt-6/modules/qtbase.nix

```diff
@@ -257,6 +258,13 @@ stdenv.mkDerivation rec {
 
   dontWrapQtApps = true;
 
+  passthru = {
+    plugins = runCommand "${finalAttrs.finalPackage.name}-only-plugins" { } ''
+      mkdir -p "$out/lib/qt-6/plugins"
+      ln -s "${finalAttrs.finalPackage}/lib/qt-6/plugins" "$out/lib/qt-6/plugins"
```

Comment:
I don't like this. It's basically just here for wrapQtAppsHook?

Reply:
For now yep unless others have more ideas for their use

Reply:
The desktop environment modules could install it? Since I assume they install full `qtwayland` currently. Though I guess that would no longer be necessary.

Reply:
Desktop modules don't care if we pull in some cmake modules or not.

Reply:
I don't like this either, could have been a separate output.

Reply:
Can’t because of references to the rest of `$out`, AIUI.

Reply:
Maybe we can just fix splitting $dev properly, and then have the hooks only propagate $out?

Reply:
> Maybe we can just fix splitting $dev properly, and then have the hooks only propagate $out?

Only if there's a way to make the qt generated cmake files happy, last time I tried they blew up fantastically.

Reply:
Splitting the outputs properly seems like a good idea but are we at least in agreement that doing this `passthru` for now is a better idea than choosing between tons of programs being broken out of the box on Wayland, unwanted dependencies leaking through, or people sprinkling `qtwayland` across the tree in ways that break Darwin?

Reply:
> Maybe we can just fix splitting $dev properly, and then have the hooks only propagate $out?

Libs will still be in $out so there will still be some pollution

Reply:
I think that's fine? Most things won't be able to discover them anyway.

Reply:
`qtwayland` exposes an API for writing Wayland compositors that at least one thing in the tree uses, so we do risk undeclared dependencies if we propagate all of it.

Reply:
That's a risk we can take.

Reply:
I can move the symlinking directory in to the wrapper if this is not wanted

Reply:
> > Maybe we can just fix splitting $dev properly, and then have the hooks only propagate $out?
> 
> Only if there's a way to make the qt generated cmake files happy, last time I tried they blew up fantastically.

We used to patch cmake with for this purpose:
```diff
commit 16a7c7b31cef945a9e66e07283ed45cb8ff8e216 (HEAD -> nixpkgs)
Author: Nick Cao <nickcao@nichi.co>
Date:   Tue Apr 11 23:44:15 2023 +0800

    cmake always export install file with absolute path

diff --git a/Source/cmExportInstallFileGenerator.cxx b/Source/cmExportInstallFileGenerator.cxx
index 195737b9fc..8d7d37e6a5 100644
--- a/Source/cmExportInstallFileGenerator.cxx
+++ b/Source/cmExportInstallFileGenerator.cxx
@@ -203,7 +203,7 @@ void cmExportInstallFileGenerator::GenerateImportPrefix(std::ostream& os)
     this->IEGen->GetLocalGenerator()->GetMakefile()->GetSafeDefinition(
       "CMAKE_INSTALL_PREFIX");
   std::string const& expDest = this->IEGen->GetDestination();
-  if (cmSystemTools::FileIsFullPath(expDest)) {
+  if (true) {
     // The export file is being installed to an absolute path so the
     // package is not relocatable.  Use the configured install prefix.
     /* clang-format off */
```

Reply:
I moved the plugin symlinking to https://github.com/NixOS/nixpkgs/pull/352419